src/UI/Implementation/Component/ViewControl/Renderer.php

```diff
@@ -181,9 +184,129 @@ protected function renderSortation(Component\ViewControl\Sortation $component, R
 			->withLabel($init_label);
 
 		$tpl->setVariable('SORTATION_DROPDOWN', $default_renderer->render($dd));
-	    return $tpl->get();
+		return $tpl->get();
+	}
+
+
+	protected function renderPagination(Component\ViewControl\Pagination $component, RendererInterface $default_renderer) {
+		$tpl = $this->getTemplate("tpl.pagination.html", true, true);
+
+		$component = $component->withResetSignals();
+		$triggeredSignals = $component->getTriggeredSignals();
+		if($triggeredSignals) {
+			$signal_select = $component->getSelectSignal();
+			$signal = $triggeredSignals[0]->getSignal();
+			$options = json_encode($signal->getOptions());
+
+			$component = $component->withOnLoadCode(function($id) use ($signal_select, $signal, $options) {
+				return "
+				$(document).on('{$signal_select}', function(event, signalData) {
+
+					var triggerer = signalData.triggerer[0], 			//the shy-button
+						param = triggerer.getAttribute('data-action'), 	//the pagination-value
+						id = '{$id}',									//id of pagination control to be used
+																		//as triggerer for others
+						pagination = $('#' + id);
+
+					//trigger select-signal
+					pagination.trigger('{$signal}',
+						{
+							'id' : '{$signal}', 'event' : 'select',
+							'triggerer' : pagination,
+							'options' : options
+						}
+					);
+					return false;
+				});
```

Comment:
This is far to much JS code placed in PHP. Factor this out to a JS file add call the according function with the necessary params only here. In PHP we only want JS-One-liners.